### ISSUE TITLE
Compilation unfinished when AVX2 specified on macOS

### ISSUE THREAD

**Issue by Hikoyu** [nodchip/tanuki-#1]
Perhaps, the code below in `source/Makefile` is inappropriate.
```
else ifeq ($(TARGET_CPU),AVX2)
        CFLAGS += -DUSE_AVX2 -mbmi -mbmi2 -mavx2 -march=corei7-avx
```
I changed the code and compilation succeeded.
```
else ifeq ($(TARGET_CPU),AVX2)
        CFLAGS += -DUSE_AVX2 -mbmi -mbmi2 -mavx2 -march=core-avx2
```
Please confirm.

**Comment by nodchip** [comment]
Thank you for submitting the pull requist.
I checked the compiler options and noticed that neither "corei7-avx" nor "core-avx2" are not listed in the list.
https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html
I'm very sorry to say this one but I will reject the pull request, and investigate later.

**Comment by Hikoyu** [comment]
@nodchip Thank you for your reply.
For your information, "corei7-avx" or "core-avx2" flags were defined in gcc-4.8 or older.
https://gcc.gnu.org/onlinedocs/gcc-4.8.2/gcc/i386-and-x86-64-Options.html#i386-and-x86-64-Options
These flags did not cause any errors during compiling by both clang++-11.0.3 (included in macOS) and g++-9.3.0 (installed by HomeBrew).
